### ISSUE TITLE
Support of xs:extension and xs:include

### ISSUE THREAD

**Issue by albanm** [fnogatz/xsd2json#3]
Thanks for the impressive piece of work.

In your thesis you specifically mention that xsd2json doesn't support the extension of complex types. Do you have any project to fill this gap ? If not, do you have a solution in mind to implement this functionality that maybe another developer could contribute ?

I have no deep understanding of the problem or of your program, but I wonder if the simplest solution could be to do some merging of the complex types definitions before applying CHR rules ?

I also wonder about include and import that you say are not supported. Do you think a simple merge of included schemas in the root schema will do the trick ?

Basically: do you think a not too complex tranformation (xslt or other) applied on the schemas before processing by xsd2json could help overcome its limitations ?

Alban

**Comment by fnogatz** [comment]
Support of `xs:include` and `xs:import` has been added.

**Comment by rudda** [comment]
are there some example to use it?

**Comment by fnogatz** [comment]
The tests under `/test/xsd` contain some examples, maybe they are helpful for you?

- `test/xsd/include.xsd`
- `test/xsd/include2.xsd`
- `test/xsd/import.xsd`
